lib/src/reporting_api_parser.dart

```diff
@@ -59,16 +66,36 @@ class ReportingQueryParser {
     // Extracts the rows after header and before the empty row,
     // and then put insertion order and revenue entries into a map
     // by making ioID the key and revenue the value.
-    final revenueMap = <String, String>{};
+    final revenueMap = <String, List<InsertionOrderDailySpend>>{};
     for (final row in reportTable) {
       // skip header
       if (row[0] == 'Insertion Order ID') continue;
       // stop if reach the empty string row
       if (row[0].isEmpty) return revenueMap;
 
-      revenueMap[row[0]] = row[1];
+      final insertionOrderId = row[0];
+      final date = Util.convertStringDateToDateTime(row[1]);
+      final revenue = row[2];
+      final impression = row[3];
+
+      final dailySpendList = revenueMap.putIfAbsent(
+          insertionOrderId, () => <InsertionOrderDailySpend>[]);
+      dailySpendList.add(InsertionOrderDailySpend(date, revenue, impression));
+      revenueMap[insertionOrderId] = dailySpendList;
     }
 
     return revenueMap;
   }
 }
+
+class InsertionOrderDailySpend {
```

Comment:
this looks like a good opportunity to use a builtclass
https://pub.dev/documentation/built_value/latest/built_value/Built-class.html

its helps build data classes in dart.